quay-enterprise/quay-tectonic-identity.md

```diff
@@ -0,0 +1,79 @@
+# Quay Enterprise OIDC Auth with Tectonic Identity 
+
+This document explains how to configure Quay Enterprise to use Tectonic Identity as an OIDC provider for authorization and authentication.
+
+## Export certificate information for TLS verification of Tectonic Identity: 
+
+```
+kubectl -n tectonic-system get secret tectonic-ca-cert-secret -o json | jq -r '.data[]' | base64 -d >> tectonic-ca-ingress.crt
+kubectl -n tectonic-system get secret tectonic-ingress-tls-secret -o json | jq -r '.data["tls.crt"]' | base64 -d >> tectonic-ca-ingress.crt
+```
+
+## Upload certificate to Quay Enterprise
+
+Upload the `tectonic-ca-ingress.crt` certificate to to Quay Enterprise through the superuser panel, or follow [Adding TLS Certificates to the Quay Enterprise Container][add-certs] to upload manually.
+
+## Configure Tectonic Identity
+
+Edit the tectonic-identity configmap: 
+
+```
+kubectl -n tectonic-system edit cm tectonic-identity
+```
+
+Add an client definition under the `staticClients` section for Quay Enterprise: 
+
+```
+  - id: quay-enterprise
+    name: Quay Enterprise
+    secret: rEpLaCeThIs
+    redirectURIs:
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/attach'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/cli'
+```
+
+__secret:__ a user generated string that will later be passed to Quay Enterprise. `date | md5sum` or `openssl rand -base64 15` may be used to generate a random string. 
+
+__redirectURIs:__ callback URLs that must be passed to the tectonic-identity configmap. Replace reg.example.com with the Server Hostname of the registry. These values can be copied from the superuser panel if the oidc provider is added to Quay Enterprise before this configmap change is saved. 
```

Comment:
Tectonic Identity ConfigMap. Replace `reg.example.com` with the server hostname of the registry. ... OIDC provider ... ConfigMap ...

Reply:
Not changing server hostname as this is to match the value in the qe super user panel.